### PR TITLE
Add Compiler API related updates in 2201.2.0 release note

### PR DESCRIPTION
**Pull request by sanjana**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md [hunk 1]
```diff
@@ -68,6 +68,10 @@ A few backward-incompatible changes have been introduced during the Swan Lake Be
 
 #### New features
 
+##### Compiler API
+
+- Introduced a set of builders in the Types API which is used to construct complex types that have varying components
```
Comment (praneesha) [inline release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md:73]:
```suggestion
- Introduced a set of builders in the Types API, which are used to construct complex types that have varying components
```

**Review by dulajdilshan — APPROVED** [review]
LGTM

**Review by mohanvive — APPROVED** [review]

File: release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md [hunk 2]
```diff
@@ -68,8 +68,16 @@ A few backward-incompatible changes have been introduced during the Swan Lake Be
 
 #### New features
 
+##### Compiler API
+
+- Introduced a set of builders in the Types API, which are used to construct complex types that have varying components
```
Comment (MaryamZi) [inline release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md:73]: Are these usually listed under language changes? We don't expect the average Ballerina user to know of the semantic API, right?

Comment (dulajdilshan) [reply release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md:73]: Ah, nice catch. This should go under `Developer Tools`. Shall we change this @sanjana ?

Comment (sanjana) [reply release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md:73]: Will be fixed in #2094

**Pull request merged**